rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/transform/TbJsonPathNode.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright © 2016-2022 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.rule.engine.transform;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.PathNotFoundException;
+import com.jayway.jsonpath.spi.json.JacksonJsonNodeJsonProvider;
+import lombok.extern.slf4j.Slf4j;
+import org.thingsboard.common.util.JacksonUtil;
+import org.thingsboard.rule.engine.api.RuleNode;
+import org.thingsboard.rule.engine.api.TbContext;
+import org.thingsboard.rule.engine.api.TbNode;
+import org.thingsboard.rule.engine.api.TbNodeConfiguration;
+import org.thingsboard.rule.engine.api.TbNodeException;
+import org.thingsboard.rule.engine.api.util.TbNodeUtils;
+import org.thingsboard.server.common.data.plugin.ComponentType;
+import org.thingsboard.server.common.msg.TbMsg;
+
+import java.util.concurrent.ExecutionException;
+
+@Slf4j
+@RuleNode(
+        type = ComponentType.TRANSFORMATION,
+        name = "json path",
+        configClazz = TbJsonPathNodeConfiguration.class,
+        nodeDescription = "Extracts json element or set of elements from a message by JSONPath expression.",
+        nodeDetails = "JSONPath expression specifies a path to an element or a set of elements in a JSON structure. <br/>"
+                + "<b>'$'</b> represents the root object or array. If JSONPath expression evaluation failed, incoming "
+                + "message routes via <code>Failure</code> chain, otherwise <code>Success</code> chain is used.",
+        uiResources = {"static/rulenode/rulenode-core-config.js"},
+        icon = "functions",
+        configDirective = "tbTransformationNodeJsonPathConfig"
+)
+public class TbJsonPathNode implements TbNode {
+
+    TbJsonPathNodeConfiguration config;
+    Configuration configurationJsonPath;
+    JsonPath jsonPath;
+    String jsonPathValue;
+
+    @Override
+    public void init(TbContext ctx, TbNodeConfiguration configuration) throws TbNodeException {
+        this.config = TbNodeUtils.convert(configuration, TbJsonPathNodeConfiguration.class);
+        this.jsonPathValue = config.getJsonPath();
+        if (this.jsonPathValue != "$") {
```

Comment:
I think that it's better to add equals here and make it reverse to handle null values

Reply:
"$" can be added to the constant in the configuration class